src/dsp_tools/models/input_error.py

```diff
@@ -0,0 +1,131 @@
+from dataclasses import dataclass
+from typing import Protocol
+
+# pylint: disable=too-few-public-methods
+
+
+separator = "\n\t"
+list_separator = "\n\t- "
+
+
+class Problem(Protocol):
+    """Information about input errors."""
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+
+
+@dataclass(frozen=True)
+class DockerNotStartedProblem:
+    """This class contains information if the docker program is not started."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class DirectoryProblem:
+    """This class contains information if a path does not exist."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class FileNotFoundProblem:
+    """This class contains information if a file is not found at the expected location."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class ExcelStructureProblem:
+    """This class contains information if the structure (column names, etc.) is incorrect."""
+
+    user_msg: str
```

Comment:
I wonder if in some cases, the user message could actually be stored on the problem class, rather than being passed and stored on the instance, if the message is always the same. What do you think?

Reply:
The reason why I did not do that is because I wanted to keep this class fairly generic. The required structure of the excel is very different when we look at the lists and resources, so the same message won't work. Do you think we should create specific classes for each?

Reply:
To me, this sounds like the more precise approach, but I'm honestly not sure it's worth the effort. What do you prefer? We could also consider this change as a second step